### DISCUSSION TITLE
Best Backup Solution for Mirth XML?

### DISCUSSION THREAD

**Discussion by rjkroll** [nextgenhealthcare/connect#5374] [Q&A] [answered]
Hi,

I was just wondering what everyone else uses for backing up their Mirth channels/installations/etc.

Currently, we are using a channel found in the forums that exports all the channels and ZIP's it up to a folder, but I was just trying to see if anyone had any other solutions they used?    Does anyone try to check their channels into something like GIT or some other Version control?    Are being just backing up their whole servers with images?   What would you do if your server blew up and needed to be rebuilt?

Thanks in advance,

**Comment by pacmano1** [comment]
"Best" is somewhat subjective.   

In most of our installs we use the CLI via a cronjob for daily full backups.  No windows here though.  See https://github.com/pacmano1/DiridiumTechnologies/blob/main/mirth_bash_backup.  

Mirth's db is also backed up of course.   All engines are postgres, most via AWS RDS.  


I think quite a few people use the code template you mentioned.   

Source code control is a related but different topic of course.  All of our engagements require the core pack which had the sanity saving channel history plugin.

**Comment by tonygermano** [comment]
I'm also not going to comment on "best" but this is another tool that I know some people use https://www.saga-it.com/tech-talk/2019/03/15/mirthsync+installation+and+basic+usage

**Comment by pacmano1** [comment, marked as answer]
Saga ITs Excellent Mirth Sync: https://www.saga-it.com/tech-talk/2019/03/15/mirthsync+installation+and+basic+usage
@odoodo awesome code template: https://forums.mirthproject.io/forum/mirth-connect/support/16298-code-template-automatic-mirth-backup
@pacmano1 's  bash script: https://github.com/pacmano1/Mirth-Snippets/blob/main/mirth_bash_backup.sh
